src/App.tsx

```diff
@@ -15,49 +20,98 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+interface SortParams<T> {
+  sortReverse: boolean;
+  sortField: keyof T | '';
+}
+
+function getPreparedGoods<T>(
+  goods: T[],
+  { sortReverse, sortField }: SortParams<T>,
+) {
+  const preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods.sort((element1, element2) => {
+      const value1 = element1[sortField];
+      const value2 = element2[sortField];
+
+      if (typeof value1 === 'number' && typeof value2 === 'number') {
+        return value1 - value2;
+      }
+
+      if (typeof value1 === 'string' && typeof value2 === 'string') {
+        return value1.localeCompare(value2);
+      }
+
+      return 0;
+    });
+  }
```

Comment:
this sorting method works incorrectly for sorting alphabetically 

https://github.com/mate-academy/react_list-of-goods-hooks/assets/117587463/9452ffc2-a5bd-4cc5-b48b-a7ca0a605806